pkg/mongoproxy/plugins/schema/util.go

```diff
@@ -141,6 +142,29 @@ func Mapify(d bson.D) bson.M {
 	return m
 }
 
+// Map creates a map from the elements of the D with operator
+// It makes additional process for arrays
+func MapifyWithOp(d bson.D, m bson.M) bson.M {
+	for _, e := range d {
+		e := processArray(e)
+		itemType := fmt.Sprint(reflect.TypeOf(e.Value))
+		if itemType == "primitive.D" {
+			itemValueSet := e.Value.(bson.D).Map()
+			if val, ok := itemValueSet["$each"]; ok {
+				m[e.Key] = val
+			} else {
+				m[e.Key] = e.Value
```

Comment:
nit: to improve readability and save indents, we can put `continue` here and move default `m[e.Key] = e.Value` to end of for loop

Reply:
fixed